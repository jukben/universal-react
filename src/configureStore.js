import { createEpicMiddleware } from 'redux-observable';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'remote-redux-devtools';
import { createLogger } from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import configureEpics from './configureEpics';

import viewer from './ducks/viewer';

const commonReducers = { viewer };

const configureStore = (options, storage) => {
  const { initialState, platformDeps = {}, platformEpics = [], platformReducers = {} } = options;

  const rootEpic = configureEpics({ ...platformDeps }, platformEpics);

  const reducers = combineReducers({
    ...commonReducers,
    ...platformReducers,
  });

  const middlewares = [createEpicMiddleware(rootEpic)];

  if (process.env.NODE_ENV === 'development') {
    middlewares.push(createLogger({ diff: true, collapsed: true }));
  }

  const enhancers = composeWithDevTools(applyMiddleware(...middlewares), autoRehydrate());

  const store = createStore(reducers, initialState, enhancers);

  // let's the magic happens :â€“)
  persistStore(store, { blacklist: ['ui'], storage }).purge();

  return store;
};

export default configureStore;
