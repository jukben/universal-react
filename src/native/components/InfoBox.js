// @flow
import React from 'react';
import styled from 'styled-components/native';
import { Icon } from 'native-base';

type InfoBoxProps = {
  danger?: boolean, // eslint-disable-line
  text: string,
};

const InfoBoxWrapper = styled.View`
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 10px;
  padding-right: 10px;
  margin: 10px;
  border-style: solid;
  border-color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')};
  border-width: 1;
  align-items: center;
  flex-direction: row;
`;

const InfoBoxText = styled.Text`
  color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')}
`;

const InfoBox = ({ danger = false, text }: InfoBoxProps) =>
  (<InfoBoxWrapper danger={danger}>
    <Icon
      style={{ marginRight: 5, color: danger ? '#d73a49' : '#0366d6' }}
      name={danger ? 'hand' : 'information-circle'}
    />
    <InfoBoxText danger={danger}>
      {text}
    </InfoBoxText>
  </InfoBoxWrapper>);

export default InfoBox;
