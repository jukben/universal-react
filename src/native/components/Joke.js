// @flow
import React from 'react';
import styled from 'styled-components/native';
import { Text } from 'native-base';

const JokeWrapper = styled.View`
  padding: 5;
`;

type JokeProps = {
  text: string,
};

const Joke = ({ text }: JokeProps) => <JokeWrapper><Text>{text}</Text></JokeWrapper>;

export default Joke;
