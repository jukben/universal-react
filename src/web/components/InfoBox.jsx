// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  danger?: boolean, //eslint-disable-line
  children?: string, //eslint-disable-line
};

const InfoBoxStyled = styled.div`
  padding: 10px;
  font-family: Arial;
  border: solid 1px ${({ danger }) => (danger ? '#d73a49' : '#0366d6')}
  color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')};
  border-radius: 3px;
`;

const InfoBox = ({ danger = false, children }: Props) => (
  <InfoBoxStyled danger={danger}>
    {danger
      ? <span role="img" aria-label="stop">✋</span>
      : <span role="img" aria-label="points up">☝</span>}
    {' '}
    {children}
  </InfoBoxStyled>
);

export default InfoBox;
