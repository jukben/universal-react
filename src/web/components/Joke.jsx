// @flow

import React from 'react';
import styled from 'styled-components';

const JokeWrapper = styled.div`
  padding: 5;
`;

type JokeProps = {
  text: string,
};

const Joke = ({ text }: JokeProps) => <JokeWrapper>{text}</JokeWrapper>;

export default Joke;
