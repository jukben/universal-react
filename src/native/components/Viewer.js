// @flow
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import { Button, Spinner, Text } from 'native-base';
import { getJoke, isSearchLoading, getLastError } from '../../selectors/viewer';
import { fetchInitAction } from '../../ducks/viewer';
import type { Error } from '../../flow';

import Joke from './Joke';
import InfoBox from './InfoBox';

const Content = styled.View`
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const Wrapper = styled.View`
  margin: 20px;
`;

type ViewerProps = {
  joke: string,
  isLoading: boolean,
  fetchNext: typeof fetchInitAction,
  lastError: Error,
};

const Viewer = ({ joke, isLoading, fetchNext, lastError }: ViewerProps) =>
  (<Content>
    <Wrapper>
      {joke && <Joke text={joke} />}
      {lastError && <InfoBox danger text={lastError.error_description} />}
    </Wrapper>
    <Wrapper>
      {(joke || lastError) &&
        <Button onPress={fetchNext}>
          {isLoading
            ? <Spinner style={{ alignSelf: 'center' }} color="blue" />
            : <Text>Next one!</Text>}
        </Button>}
    </Wrapper>
  </Content>);

export default connect(
  state => ({
    joke: getJoke(state),
    isLoading: isSearchLoading(state),
    lastError: getLastError(state),
  }),
  {
    fetchNext: fetchInitAction,
  },
)(Viewer);
