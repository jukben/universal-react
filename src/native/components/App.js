// @flow

import React from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';
import { Container, Header, Spinner, Body, Title } from 'native-base';
import { appIsRunning } from '../../selectors/viewer';

import Viewer from './Viewer';

type AppProps = {
  appLoading: boolean,
};

const App = ({ appLoading }: AppProps) => (
  <Container>
    {appLoading && <Spinner style={{ flex: 1, alignSelf: 'center' }} color="blue" />}
    {!appLoading &&
      <View style={{ flex: 1 }}>
        <Header>
          <Body>
            <Title>Universal React</Title>
          </Body>
        </Header>
        <Viewer />
      </View>}
  </Container>
);

export default connect(state => ({
  appLoading: !appIsRunning(state),
}))(App);
