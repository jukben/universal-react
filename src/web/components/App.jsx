// @flow

import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import styled from 'styled-components';
import { appIsRunning } from '../../selectors/viewer';
import media from '../../common/helpers/media';
import Viewer from './Viewer';

const Content = styled.div`
  padding: 10px;
  max-width: 500px;
  margin: 10px auto;
  ${media.tablet`max-width: 100%;`}
`;

const AppLoader = styled.div`
  width: 100%;
  height: calc(100vh - 40px);
  font-family: Arial;
  font-size: 25px;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center; 
  align-items: center; 
`;

type AppProps = {
  appLoading: boolean,
};

const App = ({ appLoading }: AppProps) =>
  (<Router>
    <Content>
      {appLoading && <AppLoader>App is loading...</AppLoader>}
      {!appLoading &&
        <div>
          <Route path="/" component={Viewer} />
        </div>}
    </Content>
  </Router>);

export default connect(state => ({
  appLoading: !appIsRunning(state),
}))(App);
