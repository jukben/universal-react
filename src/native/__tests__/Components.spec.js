import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../configureStore';

import storeFetchedMock from '../../__test__/state.fetched.json';

import App from '../components/App';
import InfoBox from '../components/InfoBox';
import Joke from '../components/Joke';
import Viewer from '../components/Viewer';

const storeEmpty = configureStore({
  initialState: {},
  platformDeps: {},
  platformEpics: [],
  platformReducers: {},
});

const storeFetched = configureStore({
  initialState: storeFetchedMock,
  platformDeps: {},
  platformEpics: [],
  platformReducers: {},
});

const injected = (Component, store = storeEmpty) => <Provider store={store}>{Component}</Provider>;

describe('Web components with empty store', () => {
  it('App renders correctly', () => {
    const tree = renderer.create(injected(<App />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Viewer renders correctly', () => {
    const tree = renderer.create(injected(<Viewer />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Joke renders correctly', () => {
    const tree = renderer.create(injected(<Joke />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox (danger) with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox danger>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Web components with mocked store (fetched data)', () => {
  // CI fails on this ("shadowColor": undefined, "shadowOffset": undefined,..)
  it.skip('App renders correctly', () => {
    const tree = renderer.create(injected(<App />, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Viewer renders correctly', () => {
    const tree = renderer.create(injected(<Viewer />, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Joke renders correctly', () => {
    const tree = renderer.create(injected(<Joke />, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox>Text</InfoBox>, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox (danger) with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox danger>Text</InfoBox>, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
