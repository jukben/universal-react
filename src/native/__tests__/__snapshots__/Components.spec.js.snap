// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web components with empty store App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <ActivityIndicator
    animating={true}
    color="blue"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "height": 80,
      }
    }
    virtual={undefined}
  />
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Web components with empty store InfoBox (danger) with text renders correctly 1`] = `
<View
  danger={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#d73a49",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#d73a49",
          "fontSize": 30,
          "marginRight": 5,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    danger={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#d73a49",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Web components with empty store InfoBox with text renders correctly 1`] = `
<View
  danger={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#0366d6",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#0366d6",
          "fontSize": 30,
          "marginRight": 5,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    danger={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#0366d6",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Web components with empty store Joke renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    virtual={undefined}
  />
</View>
`;

exports[`Web components with empty store Viewer renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Web components with mocked store (not logged) App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "top": 0,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
            virtual={undefined}
          >
            Universal React
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Chuck Norris irons his trousers with them still on.
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#007aff",
              "borderColor": "#007aff",
              "borderRadius": 5,
              "borderWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
              }
            }
            virtual={undefined}
          >
            Next one!
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Web components with mocked store (not logged) InfoBox (danger) with text renders correctly 1`] = `
<View
  danger={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#d73a49",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#d73a49",
          "fontSize": 30,
          "marginRight": 5,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    danger={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#d73a49",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Web components with mocked store (not logged) InfoBox with text renders correctly 1`] = `
<View
  danger={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#0366d6",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#0366d6",
          "fontSize": 30,
          "marginRight": 5,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    danger={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#0366d6",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Web components with mocked store (not logged) Joke renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    virtual={undefined}
  />
</View>
`;

exports[`Web components with mocked store (not logged) Viewer renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        virtual={undefined}
      >
        Chuck Norris irons his trousers with them still on.
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#007aff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": null,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "lineHeight": 19,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        virtual={undefined}
      >
        Next one!
      </Text>
    </View>
  </View>
</View>
`;
